squad-server/plugins/auto-kick-unassigned.js

```diff
@@ -0,0 +1,225 @@
+import BasePlugin from './base-plugin.js';
+import Logger from 'core/logger';
+
+export default class AutoKickUnassigned extends BasePlugin {
+  static get description() {
+    return 'The <code>AutoKickUnassigned</code> plugin will automatically kick players that are not in a squad after a specified ammount of time.';
+  }
+
+  static get defaultEnabled() {
+    return true;
+  }
+
+  static get optionsSpecification() {
+    return {
+      adminList: {
+        required: true,
+        description: 'List of server admins.',
+        connector: 'remoteAdminLists',
+        default: 'remoteAdminLists'
+      },
+      warningMessage: {
+        required: false,
+        description: 'Message SquadJS will send to players warning them they will be kicked',
+        default: 'Join a squad, you are are unassigned and will be kicked'
+      },
+      kickMessage: {
+        required: false,
+        description: 'Message to send to players when they are kicked',
+        default: 'Unassigned - automatically removed'
+      },
+      frequencyOfWarnings: {
+        required: false,
+        description: 'How often in seconds should we warn the player about being unassigned?',
+        default: 30
+      },
+      unassignedTimer: {
+        required: false,
+        description: 'How long in minutes to wait before a player that is unassigned is kicked',
+        default: 6
+      },
+      playerThreshold: {
+        required: false,
+        description:
+          'Player count required for AutoKick to start kicking players to disable set to -1 to disable',
+        default: 93
+      },
+      queueThreshold: {
```

Comment:
Is this needed or just added complexity? If the server is full you probably should kick AFKs so there is no queue and people can join anyway. Using the player threshold to keep the server between 90-100 would make it look full, but not be entirely full.

Reply:
yeah, probably is I was just thinking of things you could do and never stopped to think if I should :). I've removed it.